### PR TITLE
Fixes to Semle warnings

### PR DESCRIPTION
**Pull request by adizaimi**
Signed-off-by: Adi Zaimi <azaimi@bloomberg.net>

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/947070c/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/947070c/smoketest.log'>Error &#x26A0; </a>.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>379/379 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>cldeadlock [setup failure]<br>queuedb_rollover [setup failure]<br>sirace [setup failure]<br>truncatesc [setup failure]<br>socksql_master_swings [setup failure]<br>maxtable [setup failure]<br>incremental_backup_load [setup failure]<br>systable_locking [setup failure]<br>odh_blobs [setup failure]<br>incrementalcopy [setup failure]</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/947070c/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/947070c/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2562'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c9fcc9f/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c9fcc9f/smoketest.log'>Error &#x26A0; </a>.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>379/379 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>queuedb_rollover [setup failure]<br>sirace [setup failure]<br>cldeadlock [setup failure]<br>truncatesc [setup failure]<br>socksql_master_swings [setup failure]<br>maxtable [setup failure]<br>incremental_backup_load [setup failure]<br>systable_locking [setup failure]<br>odh_blobs [setup failure]<br>incrementalcopy [setup failure]</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c9fcc9f/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c9fcc9f/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2562'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/47c4644/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>3/379 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>socksql_master_swings<br>tunables<br>queuedb_rollover</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/47c4644/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/47c4644/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2562'>Resubmit</a></p>

File: db/osqlcheckboard.c [hunk 1]
```diff
@@ -111,46 +111,46 @@ int osql_checkboard_init(void)
 void osql_checkboard_destroy(void) { /* TODO*/ }
 
 /* insert entry into checkerboard */
-static inline int insert_into_checkerboard(osql_checkboard_t *checkboard,
+static inline int insert_into_checkerboard(osql_checkboard_t *cb,
```
Comment (riverszhang89) [inline db/osqlcheckboard.c:114]: What's the warning here?

Comment (adizaimi) [reply db/osqlcheckboard.c:114]: local variable hides global variable.

Comment (riverszhang89) [reply db/osqlcheckboard.c:114]: Got it! Thank you!

**Review by riverszhang89 — APPROVED** [review]
Good to merge.

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/a616510/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>4/380 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>ruleset_no_fp_generated [core dumped]<br>tunables<br>logfill<br>sc_constraints_logicalsc_generated</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/a616510/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/a616510/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2562'>Resubmit</a></p>

**Comment by adizaimi** [comment]
Thank you Rivers.

**Pull request merged**